owrx/websocket.py

```diff
@@ -175,60 +177,67 @@ def read_loop(self):
         def protected_read(num):
             data = self.handler.rfile.read(num)
             if data is None:
-                raise Drained()
-            if len(data) != num:
+                data = bytes()
+            while self.open and len(data) < num:
+                    (read, _, _) = select.select([self.interruptPipeRecv, self.handler.rfile], [], [], 15)
+                    if self.handler.rfile in read:
+                        data += self.handler.rfile.read(num - len(data))
```

Comment:
data length should be checked after every read. if the `rfile` was in the select, but has 0 data available, this typically means the socket has been closed.